### PR TITLE
OOP School Library: Add Three Classes

### PR DESCRIPTION
**Pull request by DammyShittu**
## In this milestone, I;

- created a `Person` class that contained the following;
    - Instance vars
    - Constructor
    - Getters and Setters
    - Private and Public Methods
    
- created a `Student` class that;
    - inherited from the `Person` class,
    - added `@classroom` to the constructor, and
    - created `play_hooky` method
    
- created a `Teacher` class that;
    - inherited from the `Person` class,
    - added `@specialization` to the constructor, and
    - overrode `can_use_services?`.
    
- saved each class in separate files.

- correctly set up linters.

## Note

Linters failed to pass without renaming `is_of_age?` to `of_age?` in `person.rb` file.

See the image below 👇

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
```
Comment (SajjadAhmad14) [inline person.rb]: Please write attr_reader and attr_accessor before the constructor of the class.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown')
+    super(age, name)
+    @classroom = classroom
```
Comment (SajjadAhmad14) [inline student.rb]: Please add `parent_permission = true` as the parameter in the constructor of `Student` class and pass it as the argument to the constructor of superclass like `super(age, name, parent_permission)`.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown')
+    super(age, name)
+    @specialization = specialization
+  end
```
Comment (SajjadAhmad14) [inline teacher.rb]: Please add `parent_permission = true` as the parameter in the constructor of `Teacher` class and pass it as the argument to the constructor of superclass like `super(age, name, parent_permission)`.

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    if of_age? || @parent_permission
+      true
+    else
+      false
+    end
```
Comment (SajjadAhmad14) [inline person.rb]: [ OPTIONAL] You may remove the `if else` block here and just write `of_age? || @parent_permission`  as `||` operator returns boolean value.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Dammy Shittu, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git Flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: student.rb (hunk 2 above)
Comment (DammyShittu) [reply student.rb]: Hi @SajjadAhmad14,

Thank you for the review.

However, adding `parent_permission = true` as a parameter of both the `Student` and `Teacher` class would throw an error when you're creating an new instance of the classes.

Attached is a screenshot that was taken from my editor after adding `parent_permission = true` and creating a new instance in `irb`.



Kindly check.

Thanks. 👍

**Comment by DammyShittu** [comment]
Hi @SajjadAhmad14 👋 

I have implemented all the changes that you suggested. I will proceed to request another review now.

Thanks 😁

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @DammyShittu 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**